grails-app/domain/org/pih/warehouse/stockMovement/StockMovement.groovy

```diff
@@ -264,14 +259,47 @@ class StockMovement {
         return name
     }
 
+    void createLineItems() {
+        lineItems = new ArrayList()
+
+        if (order) {
+            if (order.orderItems) {
+                order.orderItems.findAll{ it.orderItemStatusCode != OrderItemStatusCode.CANCELED && it.getQuantityRemainingToShip() > 0 }.each { orderItem ->
+                    StockMovementItem stockMovementItem = StockMovementItem.createFromOrderItem(orderItem)
+                    stockMovementItem.sortOrder = lineItems ? lineItems.size() * 100 : 0
+                    lineItems.add(stockMovementItem)
+                }
+            }
+        } else if (requisition) {
+            if (requisition.requisitionItems) {
+                SortedSet<RequisitionItem> requisitionItems = new TreeSet<RequisitionItem>(requisition.requisitionItems)
+                requisitionItems.each { requisitionItem ->
+                    if (!requisitionItem.parentRequisitionItem) {
+                        StockMovementItem stockMovementItem = StockMovementItem.createFromRequisitionItem(requisitionItem)
+                        lineItems.add(stockMovementItem)
+                    }
+                }
+            }
+        } else if (shipment && shipment.shipmentItems) {
+            shipment.shipmentItems.each { ShipmentItem shipmentItem ->
+                StockMovementItem stockMovementItem = StockMovementItem.createFromShipmentItem(shipmentItem)
+                if (!stockMovementItem.sortOrder) {
+                    stockMovementItem.sortOrder = lineItems ? lineItems.size() * 100 : 0
+                }
+
+                lineItems.add(stockMovementItem)
+            }
+        }
+    }
+
     static StockMovement createFromShipment(Shipment shipment) {
         return createFromShipment(shipment, Boolean.TRUE)
     }
 
     static StockMovement createFromShipment(Shipment shipment, Boolean includeStockMovementItems) {
 
-        String statusCode = (shipment.status.code == ShipmentStatusCode.PENDING) ?
-                RequisitionStatus.PENDING.toString() : RequisitionStatus.ISSUED.toString()
+        StockMovementStatusCode statusCode = (shipment.status.code == ShipmentStatusCode.PENDING) ?
+                StockMovementStatusCode.PENDING : StockMovementStatusCode.ISSUED
```

Comment:
Let's use DISPATCHED here since that's what the stock movement state machine uses.